### PR TITLE
[DOC] Update Tail based sampling doc for Grafana Agent

### LINKED ISSUE #2455: [DOC] Tail-based sampling not working
I reference this doc https://grafana.com/docs/tempo/latest/configuration/grafana-agent/tail-based-sampling/
1. doc is old
current config need like this
```
      tail_sampling:
        decision_wait: 10s
        num_traces: 100
        expected_new_traces_per_sec: 10
        policies:
          [
            {
              name: errors-policy,
              type: status_code,
              status_code: { status_codes: [ERROR] },
            }
          ]
```

2. don't work 
I use otel/opentelemetry-collector-contrib:0.76.1
with same tail_sampling config
sampling can work 
use grafana agent is not 
grafana agent version 
```
root@grafana-agent:/# grafana-agent --version 
agent, version main-d9462f2 (branch: main, revision: d9462f27)
  build user:       root@buildkitsandbox
  build date:       2023-05-05T22:32:14Z
  go version:       go1.20.3
  platform:         linux/amd64
  tags:             netgo,builtinassets,promtail_journal_enabled
```

### PR DESCRIPTION
**Pull request by knylander-grafana**
**What this PR does**:

We've updated the tail-based sampling doc to include a River configuration file example for Agent Flow mode. 

**Which issue(s) this PR fixes**:
Fixes #2455

**Checklist**
- [ ] Tests updated
- [X] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/sources/tempo/configuration/grafana-agent/tail-based-sampling.md [hunk 1]
```diff
@@ -50,194 +57,142 @@ If you're using a multi-instance deployment of the agent,
...
+        // This block determines the total length of the trace in milliseconds.
+        latency {
+            threshold_ms = 200
+        }
+    }
+
+    // The output block forwards the kept traces onto the batch processor, which will marshall them
+    // for exporting to Tempo.
+    output {
+        traces = [otelcol.processor.batch.default.input]
```
Comment (hedss) [inline docs/sources/tempo/configuration/grafana-agent/tail-based-sampling.md:128]:
```suggestion
        traces = [otelcol.exporter.otlp.tempo.input]
```
This was wrong, as we don't actually include a batch processor in the example. Have corrected.

File: docs/sources/tempo/configuration/grafana-agent/tail-based-sampling.md [hunk 2]
```diff
@@ -50,194 +57,142 @@ If you're using a multi-instance deployment of the agent,
...
+
+```river
+otelcol.receiver.otlp "otlp_receiver" {
+    grpc {
+        endpoint = "0.0.0.0:4317"
+    }
+
+    output {
+        traces = [
+            otelcol.processor.tail_sampling.errors.input,
```
Comment (hedss) [inline docs/sources/tempo/configuration/grafana-agent/tail-based-sampling.md:75]:
```suggestion
            otelcol.processor.tail_sampling.policies.input,
```

File: docs/sources/tempo/configuration/grafana-agent/tail-based-sampling.md [hunk 3]
```diff
@@ -50,194 +57,142 @@ If you're using a multi-instance deployment of the agent,
...
+
+otelcol.exporter.otlp "tempo" {
+    client {
+        endpoint = "tempo:4317"
+    }
+}
+
+// The Tail Sampling processor will use a set of policies to determine which received traces to keep
+// and send to Tempo.
+otelcol.processor.tail_sampling "errors" {
```
Comment (hedss) [inline docs/sources/tempo/configuration/grafana-agent/tail-based-sampling.md:88]:
```suggestion
otelcol.processor.tail_sampling "policies" {
```

Comment (hedss) [reply docs/sources/tempo/configuration/grafana-agent/tail-based-sampling.md:88]: We started with just errors but added a latency policy too, so have updated this.

**Review by hedss — CHANGES_REQUESTED** [review]
We missed these when writing them yesterday, should update.

File: docs/sources/tempo/configuration/grafana-agent/tail-based-sampling.md [hunk 4]
```diff
@@ -50,194 +57,142 @@ If you're using a multi-instance deployment of the agent,
...
+    num_traces = 100
+
+    // The following policies follow a logical OR pattern, meaning that if any of the policies match,
+    // the trace will be kept. For logical AND, you can use the `and` policy. Every span of a trace is
+    // examined by each policy in turn. A match will cause a short-circuit.
+
+    // This policy defines that traces that contain errors should be kept.
+    policy {
+        // The name of the policy can be used for logging purposes.
+        name = "only-sample-erroring-traces"
```
Comment (hedss) [inline docs/sources/tempo/configuration/grafana-agent/tail-based-sampling.md:102]:
```suggestion
        name = "sample-erroring-traces"
```

File: docs/sources/tempo/configuration/grafana-agent/tail-based-sampling.md [hunk 5]
```diff
@@ -50,194 +57,142 @@ If you're using a multi-instance deployment of the agent,
...
+        // in this case the OpenTelemetry 'ERROR' code.
+        status_code {
+            status_codes = [ "ERROR" ]
+        }
+    }
+
+    // This policy defines that only traces that are longer than 200ms in total should be kept.
+    policy {
+        // The name of the policy can be used for logging purposes.
+        name = "only-sample-long-traces"
```
Comment (hedss) [inline docs/sources/tempo/configuration/grafana-agent/tail-based-sampling.md:116]:
```suggestion
        name = "sample-long-traces"
```

**Review by hedss — CHANGES_REQUESTED** [review]

File: docs/sources/tempo/configuration/grafana-agent/tail-based-sampling.md [hunk 6]
```diff
@@ -50,194 +57,142 @@ If you're using a multi-instance deployment of the agent,
...
+    }
+}
+
+// The Tail Sampling processor will use a set of policies to determine which received traces to keep
+// and send to Tempo.
+otelcol.processor.tail_sampling "errors" {
+    // Total wait time from the start of a trace before making a sampling decision. Note that smaller time
+    // periods can potentially cause a decision to be made before the end of a trace has occurred.
+    decision_wait = "30s"
+    // The number of traces to keep in memory by the Agent.
```
Comment (hedss) [inline docs/sources/tempo/configuration/grafana-agent/tail-based-sampling.md:92]: Get rid of lines 92 and 93. There's actually something that can go wrong here if there's a high latency.

**Review by hedss — CHANGES_REQUESTED** [review]

File: docs/sources/tempo/configuration/grafana-agent/tail-based-sampling.md [hunk 7]
```diff
@@ -50,194 +57,142 @@ If you're using a multi-instance deployment of the agent,
...
+    num_traces = 100
```
Comment (knylander-grafana) [inline docs/sources/tempo/configuration/grafana-agent/tail-based-sampling.md:93]:
```suggestion
```

**Comment by knylander-grafana** [comment]
thank you for the review, @hedss. I've merged your suggestions.

File: docs/sources/tempo/configuration/grafana-agent/tail-based-sampling.md [hunk 8]
```diff
@@ -50,194 +57,140 @@ If you're using a multi-instance deployment of the agent,
...
+    }
+}
+
+otelcol.exporter.otlp "tempo" {
+    client {
+        endpoint = "tempo:4317"
+    }
+}
+
+// The Tail Sampling processor will use a set of policies to determine which received traces to keep
```
Comment (hedss) [inline docs/sources/tempo/configuration/grafana-agent/tail-based-sampling.md:86]: Noticed that some of these comment lines overflow in code mode, so you can't read the end of them, we should make them shorter and add more comment lines.

**Review by hedss — CHANGES_REQUESTED** [review]
Only a few things, but we need to shorten the comment lines!

File: docs/sources/tempo/configuration/grafana-agent/_index.md [hunk 9]
```diff
@@ -23,13 +23,25 @@ On top of receiving and exporting traces, the Grafana Agent contains many
 features that make your distributed tracing system more robust, and
 leverages all the data that is processed in the pipeline.
 
+## Agent modes
+
+Grafana Agent is available in three different variants:
+
+* [Static mode](/docs/agent/latest/static): The original Grafana Agent.
+* [Static mode Kubernetes operator](/docs/agent/latest/operator): The Kubernetes operator for Static mode.
```
Comment (joe-elliott) [inline docs/sources/tempo/configuration/grafana-agent/_index.md:31]: my impression is that the agent team is looking to deprecate the k8s operator. let's remove this reference.

**Review by joe-elliott — APPROVED** [review]

File: docs/sources/tempo/configuration/grafana-agent/_index.md (hunk 9 above)
Comment (knylander-grafana) [inline docs/sources/tempo/configuration/grafana-agent/_index.md:31]:
```suggestion
Grafana Agent is available in two different variants:

* [Static mode](/docs/agent/latest/static): The original Grafana Agent.
```

Comment (knylander-grafana) [reply docs/sources/tempo/configuration/grafana-agent/_index.md:31]: Done.

File: docs/sources/tempo/configuration/grafana-agent/tail-based-sampling.md (hunk 8 above)
Comment (knylander-grafana) [reply docs/sources/tempo/configuration/grafana-agent/tail-based-sampling.md]: Fixed.

**Comment by knylander-grafana** [comment]
> We missed these when writing them yesterday, should update.

Fixed the lined length.

**Comment by knylander-grafana** [comment]
> Only a few things, but we need to shorten the comment lines!

Done.

**Comment by knylander-grafana** [comment]
@hedss  I think we've addressed and discussed the changes you requested. i shortened the code comments so they no longer cause scrolling in the page preview for the River example. We should be good to merge, unless you had anything else?

**Review by hedss — APPROVED** [review]

**Pull request merged**